### ISSUE TITLE
read-package-tree 5.3.0 broke expectations of package.

### ISSUE THREAD

**Issue by CyrusNajmabadi** [npm/read-package-tree#21]
We use `read-package-tree` in https://github.com/pulumi/pulumi.  The latest version ended up breaking us due to full paths being returned where relative paths were previously returned.  Our PR that fixed the issue was here, where we lock to 5.2:

https://github.com/pulumi/pulumi/pull/2882

Previously, when running read-package-tree on a package.json file like:

```json
{
    "name": "config_basic_js",
    "comment": "The version flag below is invalid.  But we don't want to throw on it.",
    "version": "0.1",
    "license": "Apache-2.0",
    "main": "bin/index.js",
    "scripts": {
        "build": "tsc"
    },
    "optionalDependencies": {
        "typescript": "^2.5.3"
    },
    "peerDependencies": {
        "pulumi": "latest"
    },
    "dependencies": {
        "@types/node": "^9.4.7"
    }
}
```

We would get back values like so: `["node_modules/@types/node","node_modules/typescript"]` for the values in the root-node's `.package.dependencies` object.  however, with 5.3 we started getting values like:

`["/tmp/p-it-travis-job-invalid_pa-16bfb8c0-464686476/node_modules/@types/node","/tmp/p-it-travis-job-invalid_pa-16bfb8c0-464686476/node_modules/typescript"]`

It doesn't seem like this change in behavior would be expected.  Was it known and intentional?  Or is this a bug in 5.3 that has been introduced and should be fixed?  Thanks!

**Comment by CyrusNajmabadi** [comment]
Tagging @isaacs .  Any thoughts here on this?  Was this change intentional?

**Comment by isaacs** [comment]
I found the same thing just today merging it into npm, and so I'm keeping it at 5.2.x for the time being.

I have a failing test locally, and in the next few days, I'll either release a 5.3.1 that fixes it, or a 5.3.1 that backs out the realpath caching changes.  Either way, 5.2.x is safe for now, 5.3.1 will be arriving shortly to address the issue.

My apologies for the inconvenience.

**Comment by isaacs** [comment]
Should  be fixed on  v5.3.1,  just published to npm.  Care to check it out?

**Comment by CyrusNajmabadi** [comment]
Sure.  Happy to try out.
